waveorder/models/phase2D_3D.py

```diff
@@ -0,0 +1,99 @@
+import torch
+import numpy as np
+from waveorder import optics, util
+
+
+def calc_TF(YX_shape, YX_ps, Z_pos_list, lamb_ill, n_media, NA_ill, NA_obj):
+    frr = util.gen_radial_freq(YX_shape, YX_ps)
+
+    ill_pupil = optics.gen_pupil(frr, NA_ill, lamb_ill)
+    det_pupil = optics.gen_pupil(frr, NA_obj, lamb_ill)
+    Hz_stack = optics.gen_Hz_stack(
+        frr, det_pupil, lamb_ill / n_media, torch.tensor(Z_pos_list)
+    )
+
+    ZYX_shape = (len(Z_pos_list),) + YX_shape
+
+    Hu = torch.zeros(ZYX_shape, dtype=torch.complex64)
+    Hp = torch.zeros(ZYX_shape, dtype=torch.complex64)
+    for z in range(len(Z_pos_list)):
+        Hu[z], Hp[z] = optics.WOTF_2D_compute(
+            ill_pupil, det_pupil * Hz_stack[z]
+        )
+
+    return Hu, Hp
+
+
+def visualize_TF(viewer, Hu, Hp, ZYX_scale):
+    # TODO: consider generalizing w/ phase3Dto3D.visualize_TF
+    arrays = [
+        (torch.imag(Hu).numpy(), "Im(H_u)"),
```

Comment:
In the extreme case (when the tensor is on a GPU and being differentiated), one would need `tensor.detach().cpu().numpy()` to get the NumPy array.

Since we won't compute gradients in the near term, `tensor.cpu().numpy()` is enough.

Reply:
Fixed, thanks!